### PR TITLE
Datastore

### PR DESCRIPTION
**Pull request by GuillermoTafoya**
Defined the classes for the database, defined methods for retrieving, deleting and updating users; and defined the connection for a login/signup form.

Register Page
image
User Info Page (For Debugging Purposes)
image

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/DeleteAll.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.DatastoreService;
+//import com.google.sps.data.Day;
```
Comment (cindyxue) [inline src/main/java/com/google/sps/servlets/DeleteAll.java]: Remove the comments?

File: src/main/java/com/google/sps/data/DatastoreService.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+//import com.google.cloud.datastore.StructuredQuery.OrderBy;
```
Comment (cindyxue) [inline src/main/java/com/google/sps/data/DatastoreService.java]: nit: remove the comment?

File: src/main/java/com/google/sps/data/DatastoreService.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            String firstName = entity.getString("firstName");
+            String middleName = entity.getString("middleName");
+            String lastName = entity.getString("lastName");
+
+            var calendar = entity.getList("calendar");
+            var journal = entity.getList("journal");
+
+            long panicButtonPressed = entity.getLong("panicButtonPressed");
+
+            User user = new User(username, email, passwordHash, firstName, middleName, lastName, calendar, journal, panicButtonPressed);
```
Comment (cindyxue) [inline src/main/java/com/google/sps/data/DatastoreService.java]: Refactor this into a method (i.e. createUser)? You can replace line 28-41 with this method too to reduce duplicate code.

File: src/main/java/com/google/sps/data/DatastoreService.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+                .set("email", user.getEmail())
+                .set("passwordHash", user.getPasswordHash())
+                .set("firstName", user.getFirstName())
+                .set("middleName", user.getMiddleName())
+                .set("lastName", user.getLastName())
+                .set("calendar", user.getCalendar())
+                .set("journal", user.getJournal())
+                .set("panicButtonPressed", user.getPanicButtonPressed())
+                .build();
+        // datastore.add(user); // Datastore add operation: inserts the provided entities.
```
Comment (cindyxue) [inline src/main/java/com/google/sps/data/DatastoreService.java]: nit: remove this?

**Review by cindyxue — COMMENTED** [review]
Thanks for working on this! Overall looks good to me!

Could you please add a screenshot of the registration page in the PR description?

File: src/main/java/com/google/sps/servlets/DeleteAll.java (hunk 1 above)
Comment (GuillermoTafoya) [reply src/main/java/com/google/sps/servlets/DeleteAll.java]: Yup, thanks. Removed innecesary commented code! :)

File: src/main/java/com/google/sps/data/DatastoreService.java (hunk 3 above)
Comment (GuillermoTafoya) [reply src/main/java/com/google/sps/data/DatastoreService.java]: Done! :D 
Wrote two methods to keep the code a bit cleaner, thanks!

**Pull request merged**